### PR TITLE
Feature/add jdbc snowflake

### PR DESCRIPTION
**Pull request by swatiarora0208**
**What this PR does / why we need it**:
This PR is to add snowflake store for feature sink using the jdbc-snowflake

### CODE REVIEW

File: storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/common/JdbcTemplater.java [hunk 1]
```diff
@@ -22,6 +22,12 @@
 
 public interface JdbcTemplater extends Serializable {
   static String getTableName(FeatureSetProto.FeatureSetSpec featureSetSpec) {
+    System.out.println(
```
Comment (dr3s) [inline storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/common/JdbcTemplater.java:25]: is this the convention?  Isn't there a proper logger?

File: storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/writer/JdbcFeatureSink.java [hunk 2]
```diff
@@ -144,11 +147,23 @@ private static Connection connect(JdbcConfig config) {
     String password = config.getPassword();
     String className = config.getClassName();
     String url = config.getUrl();
-
     try {
-      Class.forName(className);
+
       if (!username.isEmpty()) {
-        return DriverManager.getConnection(url, username, password);
+        if (className == "net.snowflake.client.jdbc.SnowflakeDriver") {
```
Comment (dr3s) [inline storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/writer/JdbcFeatureSink.java:153]: eventually this should be encapsulated into a connectionprovider

File: storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/writer/JdbcWrite.java [hunk 3]
```diff
@@ -312,4 +304,28 @@ public static void setPreparedStatementValue(
...
   }
+
+  private static JdbcIO.DataSourceConfiguration create_dsconfig(
+      StoreProto.Store.JdbcConfig jdbcConfig) {
+    String username = jdbcConfig.getUsername();
+    String password = jdbcConfig.getPassword();
+    String className = jdbcConfig.getClassName();
+    String url = jdbcConfig.getUrl();
+
+    if (className == "net.snowflake.client.jdbc.SnowflakeDriver") {
```
Comment (dr3s) [inline storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/writer/JdbcWrite.java:315]: ditto

**Review by dr3s — APPROVED** [review]

File: storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/common/JdbcTemplater.java (hunk 1 above)
Comment (zhangchi1) [reply storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/common/JdbcTemplater.java:25]: We added:        
```
 <dependency>
		    <groupId>org.slf4j</groupId>
		    <artifactId>slf4j-jdk14</artifactId>
		    <version>1.7.25</version>
		</dependency>
``` 
for a better logger under jdbc pom.xml

**Pull request merged**